PWGHF/Core/HfMlResponseDplusToPiKPi.h

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file HfMlResponseDplusToPiKPi.h
+/// \brief Class to compute the ML response for D± → π± K∓ π± analysis selections
+/// \author Alexandre Bigot <alexandre.bigot@cern.ch>, IPHC Strasbourg
+
+#ifndef PWGHF_CORE_HFMLRESPONSEDPLUSTOPIKPI_H_
+#define PWGHF_CORE_HFMLRESPONSEDPLUSTOPIKPI_H_
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "PWGHF/Core/HfMlResponse.h"
+
+namespace o2::analysis
+{
+
+enum InputFeaturesDplusToPiKPi : int8_t {
+  ptProng0 = 0,
+  ptProng1,
+  ptProng2,
+  impactParameter0,
+  impactParameter1,
+  impactParameter2,
```

Comment:
Maybe we could specify `XY` on these and add those in the `Z` direction (this would also allow us to include the case of the HF triggers, for which we could move to this code as soon as stable)

Reply:
I may have missed something, but it looks like the `Z` component is not even defined in the [`HfCand3Prong` table](https://github.com/AliceO2Group/O2Physics/blob/master/PWGHF/DataModel/CandidateReconstructionTables.h#L770). In the [3prong candidate creator](https://github.com/AliceO2Group/O2Physics/blob/master/PWGHF/TableProducer/candidateCreator3Prong.cxx#L207), we only store the `XY` component via `getY` defined [here](github.com/AliceO2Group/AliceO2/blob/dev/DataFormats/Reconstruction/include/ReconstructionDataFormats/DCA.h#L57).

So, we should add the definition in the 3prong table, store `impactParameter*.getZ()` in 3 prong creator, so we can retrieve it in the Dplus selector :)

Reply:
Indeed you are right, I missed it! 
Then let's do it in a second step, together with the change in the creator(s).

Reply:
But, addig `XY` in the enum entries can already be done, it is indeed clearer this way.